### PR TITLE
Add comments to TestGetControlPlaneComponents for clarity

### PR DESCRIPTION
**Pull request by gnarayan1337**
This commit adds detailed inline comments to the TestGetControlPlaneComponents function, improving readability and clarity for future maintainers. The comments describe the purpose of each test case and explain key steps in the test logic, including:

A brief overview of the test function.
Descriptions for each test case, highlighting different configurations tested (e.g., custom ports and addresses vs. defaults). Explanations of test assertions to ensure the function output matches expected results. These comments provide context for developers reviewing or modifying the test, facilitating better understanding and easier maintenanc

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/apiclient/wait_test.go [hunk 1]
```diff
@@ -24,13 +24,16 @@ import (
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
+// TestGetControlPlaneComponents tests the getControlPlaneComponents function
 func TestGetControlPlaneComponents(t *testing.T) {
```
Comment (carlory) [inline cmd/kubeadm/app/util/apiclient/wait_test.go:28]: The function name is enough to understand, so we may not need to explain it. Other comments are not necessary, the code is self-explanatory.

Comment (carlory) [reply cmd/kubeadm/app/util/apiclient/wait_test.go:28]: Thanks for your willingness to improve the code. Welcome to the community.

Comment (gnarayan1337) [reply cmd/kubeadm/app/util/apiclient/wait_test.go:28]: No problem, are there any areas where these kind of changes could be used?

**Review by neolit123 — COMMENTED** [review]
/close

we don't accept such simple changes.
you can help by writing more unit tests.

**Pull request closed without merging**